trading_calendars/xbkk_holidays.py

```diff
@@ -0,0 +1,319 @@
+    pd.Timestamp('2015-01-02', tz=UTC),
+]
+
+asanha_bucha_bridge_days = [
+    pd.Timestamp('2009-07-06', tz=UTC),
+    pd.Timestamp('2016-07-18', tz=UTC),
+]
+
+queens_birthday_bridge_days = [
+    pd.Timestamp('2010-08-13', tz=UTC),
+    pd.Timestamp('2014-08-11', tz=UTC),
+]
+
+coronation_bridge_days = [
+    pd.Timestamp('2015-05-04', tz=UTC),
+    pd.Timestamp('2016-05-06', tz=UTC),
+]
+
+vesak_bridge_days = [
+    pd.Timestamp('2011-05-16', tz=UTC),
+]
+
+misc_adhoc = [
+    pd.Timestamp('2006-04-19', tz=UTC),  # Special Holiday
+    pd.Timestamp('2006-06-12', tz=UTC),  # Special Holiday
+    pd.Timestamp('2006-06-13', tz=UTC),  # Special Holiday
+    pd.Timestamp('2006-09-20', tz=UTC),  # Exchange Holiday
+    pd.Timestamp('2007-12-24', tz=UTC),  # Exchange Holiday
+    pd.Timestamp('2010-05-20', tz=UTC),  # Closure Due to Security Concerns
+    pd.Timestamp('2010-05-21', tz=UTC),  # Closure Due to Security Concerns
+    pd.Timestamp('2012-04-09', tz=UTC),  # Bank Holiday
+    pd.Timestamp('2017-10-26', tz=UTC),  # Cremation of King Bhumibol
+]
+
+# Lunar Holidays
+# --------------
+
+# Makha Bucha (also known as Magha Puja) is celebrated on the day of the Full
+# Moon of Magha in the Buddhist calendar. This falls sometime between February
+# and March.
+makha_bucha = pd.to_datetime([
+    '1981-02-18',
+    '1982-02-08',
+    '1983-02-27',
+    '1984-02-16',
+    '1985-03-06',
+    '1986-02-24',
+    '1987-02-13',
+    '1988-03-03',
+    '1989-02-20',
+    '1990-02-09',
+    '1991-02-28',
+    '1992-02-18',
+    '1993-03-08',
+    '1994-02-25',
+    '1995-02-15',
+    '1996-03-05',
+    '1997-02-22',
+    '1998-02-11',
+    '1999-03-02',
+    '2000-02-19',
+    '2001-02-08',
+    '2002-02-26',
+    '2003-02-17',
+    '2004-03-05',
+    '2005-02-23',
+    '2006-02-13',
+    '2007-03-05',
+    '2008-02-21',
+    '2009-02-09',
+    '2010-03-01',
+    '2011-02-18',
+    '2012-03-07',
+    '2013-02-25',
+    '2014-02-14',
+    '2015-03-04',
+    '2016-02-22',
+    '2017-02-13',
+    '2018-03-01',
+    '2019-02-19',
+])
+
+# Vesak (also known as Buddha Day) is celebrated on the day of the Full Moon of
+# Visakha in the Buddhist calendar. This typically falls in May.
+vesak = pd.to_datetime([
+    '1981-05-18',
+    '1982-05-07',
+    '1983-05-26',
+    '1984-05-15',
+    '1985-06-02',
+    '1986-05-23',
+    '1987-05-13',
+    '1988-05-31',
+    '1989-05-20',
+    '1990-05-09',
+    '1991-05-28',
+    '1992-05-16',
+    '1993-06-04',
+    '1994-05-24',
+    '1995-05-14',
+    '1996-06-01',
+    '1997-05-22',
+    '1998-05-11',
+    '1999-05-30',
+    '2000-05-18',
+    '2001-05-07',
+    '2002-05-27',
+    '2003-05-15',
+    '2004-06-02',
+    '2005-05-23',
+    '2006-05-12',
+    '2007-05-31',
+    '2008-05-19',
+    '2009-05-08',
+    '2010-05-28',
+    '2011-05-17',
+    '2012-06-04',
+    '2013-05-24',
+    '2014-05-13',
+    '2015-06-01',
+    '2016-05-20',
+    '2017-05-10',
+    '2018-05-29',
+    '2019-05-20',
+])
```

Comment:
Interesting to note the differences with [Indonesia's observance](https://github.com/quantopian/trading_calendars/pull/83/files?file-filters%5B%5D=.py#diff-b79ca87a50042e6e107c6c3567cff88fR370-R386), especially since the two countries are geographically pretty close. Calendars are hard!